besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java

```diff
@@ -0,0 +1,227 @@
+            config.getUnstable().getBonsaiTrieLogRetentionThreshold()));
+  }
+
+  static TrieLogCount getCount(
+      final BonsaiWorldStateKeyValueStorage rootWorldStateStorage,
+      final int limit,
+      final Blockchain blockchain) {
+    final AtomicInteger total = new AtomicInteger();
+    final AtomicInteger canonicalCount = new AtomicInteger();
+    final AtomicInteger forkCount = new AtomicInteger();
+    final AtomicInteger orphanCount = new AtomicInteger();
+    rootWorldStateStorage
+        .streamTrieLogKeys(limit)
+        .map(Bytes32::wrap)
+        .map(Hash::wrap)
+        .forEach(
+            hash -> {
+              total.getAndIncrement();
+              blockchain
+                  .getBlockHeader(hash)
+                  .ifPresentOrElse(
+                      (header) -> {
+                        long number = header.getNumber();
+                        final Optional<BlockHeader> headerByNumber =
+                            blockchain.getBlockHeader(number);
+                        if (headerByNumber.isPresent()
+                            && headerByNumber.get().getHash().equals(hash)) {
+                          canonicalCount.getAndIncrement();
+                        } else {
+                          forkCount.getAndIncrement();
+                        }
+                      },
+                      orphanCount::getAndIncrement);
+            });
+
+    return new TrieLogCount(total.get(), canonicalCount.get(), forkCount.get(), orphanCount.get());
+  }
+
+  private static void saveTrieLogsInFile(final Map<byte[], byte[]> trieLogs) throws IOException {
+
+    File file = new File(trieLogFile);
+
+      try (BufferedWriter bf = new BufferedWriter(new FileWriter(file, StandardCharsets.UTF_8))) {
+          for (Map.Entry<byte[], byte[]> entry : trieLogs.entrySet()) {
+              bf.write(Bytes.of(entry.getKey()) + ":" + Base64.toBase64String(entry.getValue()));
+              bf.newLine();
+          }
+          bf.flush();
+      } catch (IOException e) {
+          LOG.error(e.getMessage());
+          throw e;
+      }
+  }
+
+  private static IdentityHashMap<byte[], byte[]> readTrieLogsFromFile() throws IOException {
+
+    File file = new File(trieLogFile);
+    IdentityHashMap<byte[], byte[]> trieLogs = new IdentityHashMap<>();
+      try (BufferedReader br = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+          String line;
+          while ((line = br.readLine()) != null) {
+              List<String> parts = Splitter.on(':').splitToList(line);
+              byte[] key = Bytes.fromHexString(parts.get(0)).toArrayUnsafe();
+              byte[] value = Base64.decode(parts.get(1));
+              trieLogs.put(key, value);
+          }
+      } catch (IOException e) {
+          LOG.error(e.getMessage());
+          throw e;
+      }
+
+    return trieLogs;
+  }
+
+  private static void deleteTrieLogFile() {
+    File file = new File(trieLogFile);
```

Comment:
When we create the file we could use the File.deleteOnExit so we don't need to worry about manually deleting this file